model_analyzer/config/generate/optuna_run_config_generator.py

```diff
@@ -451,6 +455,44 @@ def _create_objective_based_run_config(
         return run_config
 
     def _create_parameter_combo(
+        self,
+        model: ModelProfileSpec,
+        trial_objectives: TrialObjectives,
+        composing_trial_objectives: ComposingTrialObjectives,
+    ) -> ParameterCombo:
+        if model.is_ensemble():
+            param_combo = self._create_ensemble_parameter_combo(
+                composing_trial_objectives
+            )
+        else:
+            param_combo = self._create_non_ensemble_parameter_combo(trial_objectives)
+
+        return param_combo
+
+    def _create_ensemble_parameter_combo(
```

Comment:
I think a code smell is having a bunch of variables or functions for each type. create_X_parameter_combo, where X is ensemble vs bls vs multi-model vs other, and then having a bunch of embedded if/else all over the code. This is really where a base class + implementation classes for each type + a factory would come in handy. Consider that for future refactors.

Reply:
Will do.